R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+#' @param population Population used to calculate the `simulationResults`
+#'   (optional). This is used only to add the population covariates to the
+#'   resulting dataframe.
+#' @param dataSets Instance (or a `list` of instances) of the `DataSet`
+#'   class.
+#' @param groups A string or a list of strings assigning the data set to a
+#'   group. If an entry within the list is `NULL`, the corresponding data set is
+#'   not assigned to any group (and the corresponding entry in the dataframe
+#'   will be an `NA`). If provided, `groups` must have the same length as
+#'   `dataSets` and/or `simulationResults`. If no grouping is specified for any
+#'   of the dataset, the column `group` in the dataframe output will be all
+#'   `NA`.
+#' @param names A string or a list of string. This argument will be encountered
+#'   across different methods:
+#'
+#' - `$setDataTransformations()`: In the context of this method, a list of names
+#' specifying which observed datasets and/or paths in simulated dataset to
+#' transform with the specified transformations. Default is `NULL`, i.e., the
+#' transformations, if any specified, will be applied to all rows of the
+#' dataframe.
+#'
+#' - `$addSimulationResults()`: In the context of this method, a list of strings
+#' assigning new names to the quantities or paths present in the entered
+#' `SimulationResults` object. Note that the datasets whose names you wish to
+#' not change should be specified as `NULL` in the list.
+#'
+#' - `$addDataSets()`: In the context of this method, a list of strings
+#' assigning new names to the list of instances of the `DataSet` class. Note
+#' that the datasets whose names you wish to not change should be specified as
+#' `NULL` in the list.
+#'
+#'
+#' @import tidyr
+#' @import ospsuite.utils
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## constructor method ---------------
+
+    #' @description
+    #' Initialize a new instance of the class.
+    #'
+    #' @return A new (empty) instance of `DataCombined` class.
+
+    # empty constructor
+    initialize = function() {
+    },
+
+    ## setter methods ---------------
+
+    #' @description
+    #' Adds observed data.
+    #' @return `DataCombined` object containing observed data.
+
+    addDataSets = function(dataSets, names = NULL, groups = NULL) {
```

Comment:
Since you are the second person who has said this, I have changed how parameters are specified now.

Only the common parameters live before the class definition, while others before class methods.